server/server.go

```diff
@@ -713,16 +717,23 @@ func (s *Server) registerHandler(router *mux.Router, version int, path string, m
 }
 
 func (s *Server) reload(ctx context.Context, txn storage.Transaction, event storage.TriggerEvent) {
+	// reset some cached info
+	s.partials = map[string]rego.PartialResult{}
+	s.revisions = map[string]string{}
 
-	if revision, err := manifest.ReadBundleRevision(ctx, s.store, txn); err != nil {
-		if !storage.IsNotFound(err) {
+	// read all bundle revisions from storage (if any exist)
+	names, err := manifest.ReadBundleNames(ctx, s.store, txn)
+	if err != nil && !storage.IsNotFound(err) {
+		panic(err)
```

Comment:
Could this be a follow-up task? It's harsh to panic... 🤔 OTOH there might be nothing better to do 🤷‍♂

Reply:
I spent some time staring at that too, unfortunately I think its the best option (right now). We could maybe retry or something but I don't think we would always want to retry blindly on any error, and we don't know what errors might cause this (IMO this error handling is more of a hypothetical than a thing that really happens).